### DISCUSSION TITLE
How to build Arduino Library from Component

### DISCUSSION THREAD

**Discussion by beriberikix** [espressif/arduino-esp32#7627] [Q&A]
We have an [ESP-IDF component](https://github.com/golioth/golioth-firmware-sdk/tree/main/examples/esp_idf) that we would like to make available to Arduino users. During my research I found [ESP32 Arduino Lib Builder](https://github.com/espressif/esp32-arduino-lib-builder) which I believe is a tool intended for that purpose. However, I'm unclear how to use it even after reviewing the documentation.

Does anyone have experiencing building a custom Arduino library based off an esp-idf component and could share any pointers?

Thanks!

**Comment by VojtechBartoska** [comment]
Hello @beriberikix,

you can take a look on [this talk](https://www.youtube.com/watch?v=aPXvIInTi4s&list=PLOzvoM7_Knrc6o-n25jYuXRB2T8UKk1NU&index=14&ab_channel=EspressifSystems) which was presented by our Developer @SuGlider. It should answer some of your questions.

**Reply by beriberikix** [reply]
Thanks @VojtechBartoska! It certainly helps. So if I understand it correctly, Arduino libraries that depend on an esp-idf component need to ship as a precompiled binary. So in order to create our own Library that we distribute outside of the `arduino-esp32` implementation, we should build our binary with lib builder, copy it to our own library and somehow include it in the Arduino build system?

The details are missing but is that the general direction?

**Reply by me-no-dev** [reply]
The lib-builder already includes a [script](https://github.com/espressif/esp32-arduino-lib-builder/blob/master/tools/copy-to-arduino.sh) to copy everything to a working Arduino core folder. You can get inspiration from it :)

**Reply by dlarue** [reply]
I think you have to modify your sdkconfig at the root of the build system and then copy it into the components/arduino/tools/sdk/esp32 directory so it builds that library package with the config changes you desire.
